### PR TITLE
PAASTA-17231-LogSuffix-Fix

### PR DESCRIPTION
**Pull request by lanceleih**
### Description
Bug fix when running `paasta logs` for containers inside flink instances.

### Testing Done
`make test`

### CODE REVIEW

File: paasta_tools/cli/cmds/logs.py [hunk 1]
```diff
@@ -1333,6 +1333,9 @@ def paasta_logs(args: argparse.Namespace) -> int:
         instances = None
     else:
         instances = args.instances.split(",")
+        # Check for flint instance names only, not the containers inside them
+        check_flink_instances = [x.split(".")[0] for x in instances if x.split(".")]
```
Comment (stug) [inline paasta_tools/cli/cmds/logs.py:1337]: A couple thoughts:
* Although we know that these are probably flink instances, the log suffix behavior is something that other instance types could use, too.  I think it'd be better to call this something like `instances_without_suffixes` (which I know doesn't really roll off the tongue), reference PAASTA-15311 in the comment, and only talk about flink as an example use case, if at all
* if there is no period character, `split()` will return a 1-element list, so you don't actually need the `if` part of this -- you can just access `x.split(".")[0]`

File: tests/cli/test_cmds_logs.py [hunk 2]
```diff
@@ -388,6 +388,19 @@ def test_extract_utc_timestamp_from_log_line_when_invalid_date_format():
...
+    service = "fake_service"
+    clusters = ["fake_cluster1"]
+    instance = "fake_instance.container"
+
+    with mock.patch(
+        "paasta_tools.cli.utils.verify_instances", autospec=True
+    ) as verify_instances_patch:
+        verify_instances_patch.return_value = True
+
+        assert verify_instances_patch(instance, service, clusters)
```
Comment (stug) [inline tests/cli/test_cmds_logs.py:401]: This doesn't actually test anything -- you've replaced `verify_instances` with a mock object that will always return `True`, so you could run `assert verify_intances_patch("abc", 123, False)` and it'd still return True.

I think you should patch `verify_instances` and then call `paasta_logs` and assert that `verify_instances` was called with just the instance names without the suffxies.  Let me know if you want help doing that.

**Review by stug — CHANGES_REQUESTED** [review]

File: tests/cli/test_cmds_logs.py [hunk 3]
```diff
@@ -388,6 +388,23 @@ def test_extract_utc_timestamp_from_log_line_when_invalid_date_format():
     assert not logs.extract_utc_timestamp_from_log_line(line)
 
 
+def test_paasta_log_line_passes_filter_true_for_instances_with_suffixes():
```
Comment (stug) [inline tests/cli/test_cmds_logs.py:391]: I think this name comes from the fact that you copied some of this from tests of the `logs.paasta_log_line_passes_filter` method.  We should rename this to explain it better.

File: paasta_tools/cli/cmds/logs.py [hunk 4]
```diff
@@ -1332,7 +1333,11 @@ def paasta_logs(args: argparse.Namespace) -> int:
     if args.instances is None:
         instances = None
     else:
+
```
Comment (stug) [inline paasta_tools/cli/cmds/logs.py:1336]: I think you added newlines inadvertently in a couple places.  I don't think they're really helping readability or anything, so probably worth removing them.

**Review by stug — APPROVED** [review]
looks good to me -- a couple tiny things that you can fix and ship.

File: tests/cli/test_cmds_logs.py (hunk 3 above)
Comment (lanceleih) [reply tests/cli/test_cmds_logs.py]: Renamed to `test_verify_instances_for_instances_with_suffixes()`

**Pull request merged**